presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/rule/SingleMarkDistinctToGroupBy.java

```diff
@@ -134,6 +134,8 @@ public Pattern getPattern()
         return new AggregationNode.Aggregation(
                 new FunctionCall(call.getName(), call.getWindow(), false, call.getArguments()),
                 aggregation.getSignature(),
-                Optional.empty());
+                Optional.empty(),
```

Comment:
The result of a DISTINCT xxxx ORDER BY yyy could be non-deterministic. But I didn't find anything saying it's not allowed. If people think this should be banned, I'm happy to remove it. @martint